pkg/tfbridge/info.go

```diff
@@ -99,39 +99,47 @@ type ProviderInfo struct {
 	PreConfigureCallbackWithLogger PreConfigureCallbackWithLogger
 }
 
+// Describe the mapping from resource and datasource tokens to Pulumi resources and
+// datasources.
+//
+// NOTE: Experimental; We are still iterating on the design of this function, and it is
+// subject to change without warning.
+type ComputeDefaultInfo struct {
+	Resource   ResourceStrategy
+	DataSource DataSourceStrategy
+}
+
 // Add mapped resources and datasources according to the given strategies.
 //
 // NOTE: Experimental; We are still iterating on the design of this function, and it is
 // subject to change without warning.
-func (info *ProviderInfo) ComputeDefaults(r ResourceStrategy, d DatasourceStrategy) error {
+func (info *ProviderInfo) ComputeDefaults(opts ComputeDefaultInfo) error {
 	var errs multierror.Error
-	err := info.ComputeDefaultResources(r)
+	err := info.computeDefaultResources(opts.Resource)
 	if err != nil {
 		errs.Errors = append(errs.Errors, fmt.Errorf("resources:\n%w", err))
 	}
-	err = info.ComputeDefaultDataSources(d)
+	err = info.computeDefaultDataSources(opts.DataSource)
 	if err != nil {
 		errs.Errors = append(errs.Errors, fmt.Errorf("datasources:\n%w", err))
 	}
 	return errs.ErrorOrNil()
 }
 
-// Apply strategy to generate missing resources.
-//
-// NOTE: Experimental; We are still iterating on the design of this function, and it is
-// subject to change without warning.
-func (info *ProviderInfo) ComputeDefaultResources(strategy ResourceStrategy) error {
+func (info *ProviderInfo) computeDefaultResources(strategy ResourceStrategy) error {
```

Comment:
Now that we have a struct for passing strategies, we never have a reason to expose these helpers. We may as well remove them during this phase.

Reply:
Yeah! I think either use a struct or remove ComputeDefaults completely and only provide the individual helpers, keep surface area a bit smaller! I think the struct is appropriate as there are the multierror helpers baked in.